tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -243,30 +216,152 @@ private static void AddAppsecSpecificInstrumentations()
 
             try
             {
-                Log.Debug("Sending CallTarget appsec derived integration definitions to native library.");
+                Log.Debug("Adding CallTarget appsec derived integration definitions to native library.");
                 var payload = InstrumentationDefinitions.GetDerivedDefinitions(InstrumentationCategory.AppSec);
                 NativeMethods.InitializeProfiler(payload.DefinitionsId, payload.Definitions);
-                foreach (var def in payload.Definitions)
-                {
-                    def.Dispose();
-                }
+                derived = payload.Definitions.Length;
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, ex.Message);
+            }
+
+            Log.Information($"{defs} AppSec definitions and {derived} derived definitions added to the profiler.");
+        }
 
-                Log.Information<int>("The profiler has been initialized with {count} AppSec derived definitions.", payload.Definitions.Length);
+        private static void RemoveAppsecSpecificInstrumentations()
+        {
+            int defs = 0, derived = 0;
+            try
+            {
+                Log.Debug("Adding CallTarget AppSec integration definitions to native library.");
+                var payload = InstrumentationDefinitions.GetAllDefinitions(InstrumentationCategory.AppSec);
+                NativeMethods.RemoveCallTargetDefinitions(payload.DefinitionsId, payload.Definitions);
+                defs = payload.Definitions.Length;
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, ex.Message);
+            }
+
+            try
+            {
+                Log.Debug("Adding CallTarget appsec derived integration definitions to native library.");
+                var payload = InstrumentationDefinitions.GetDerivedDefinitions(InstrumentationCategory.AppSec);
+                NativeMethods.RemoveCallTargetDefinitions(payload.DefinitionsId, payload.Definitions);
+                derived = payload.Definitions.Length;
             }
             catch (Exception ex)
             {
                 Log.Error(ex, ex.Message);
             }
+
+            Log.Information($"{defs} AppSec definitions and {derived} derived definitions removed from the profiler.");
         }
 
-        /// <summary>
-        /// Frees resources
-        /// </summary>
+        /// <summary> Frees resources </summary>
         public void Dispose()
         {
             _waf?.Dispose();
         }
 
+        private void FeaturesProductConfigChanged(object sender, ProductConfigChangedEventArgs e)
+        {
+            var features = e.GetDeserializedConfigurations<Features>().FirstOrDefault();
+            if (features != null)
+            {
+                _settings.Enabled = features.Asm.Enabled;
```

Comment:
I'm not sure but is this respecting the combinations specified [here ](https://datadoghq.atlassian.net/wiki/spaces/APS/pages/2466349908/Remote+Config#FEATURES---Remote-Activation)
for example, if `DD_APPSEC_ENABLED` is set to `false`, but RC says true, it should be `false` and not always override?
also this could be good case for unit tests 😅

Reply:
Yes, indeed, and it is like that. In settings there is a field called `CanBeEnabled` for that purpose. And the test `tracer/test/Datadog.Trace.Security.IntegrationTests/AspNetCore5AsmToggle.cs` checks that with the `true, false, and null` values for the DD_APPSEC_ENABLED variable (null is not set).